packages/react-mongo/react-mongo.ts

```diff
@@ -0,0 +1,206 @@
+      refs.updateOnReady = true
+      return refs.isReady
+    }
+  }
+}
+
+const useSubscriptionServer = (name?: string, ...args: any[]): Meteor.SubscriptionHandle => ({
+  stop() {},
+  ready() { return true }
+})
+
+export const useSubscription = Meteor.isServer
+  ? useSubscriptionServer
+  : useSubscriptionClient
+
+type useFindActions<T> =
+  | { type: 'refresh', data: T[] }
+  | { type: 'addedAt', document: T, atIndex: number }
+  | { type: 'changedAt', document: T, atIndex: number }
+  | { type: 'removedAt', atIndex: number }
+  | { type: 'movedTo', fromIndex: number, toIndex: number }
+
+const useFindReducer = <T>(data: T[], action: useFindActions<T>): T[] => {
+  switch (action.type) {
+    case 'refresh':
+      return action.data
+    case 'addedAt':
+      return [
+        ...data.slice(0, action.atIndex),
+        action.document,
+        ...data.slice(action.atIndex)
+      ]
+    case 'changedAt':
+      return [
+        ...data.slice(0, action.atIndex),
+        action.document,
+        ...data.slice(action.atIndex + 1)
+      ]
+    case 'removedAt':
+      return [
+        ...data.slice(0, action.atIndex),
+        ...data.slice(action.atIndex + 1)
+      ]
+    case 'movedTo':
+      const doc = data[action.fromIndex]
+      const copy = [
+        ...data.slice(0, action.fromIndex),
+        ...data.slice(action.fromIndex + 1)
+      ]
+      copy.splice(action.toIndex, 0, doc)
+      return copy
+  }
+}
+
+const useFindClient = <T = any>(factory: () => Mongo.Cursor<T>, deps: DependencyList) => {
+  const [data, dispatch] = useReducer<Reducer<T[], useFindActions<T>>, T[]>(
+    useFindReducer,
+    undefined,
+    () => Tracker.nonreactive(() => {
+      const cursor = factory()
+      return cursor.fetch()
+    })
+  )
+
+  useEffect(() => {
+    // Rebuild the cursor and data in case an update may have
+    // happened between first render and commit. Alternatively,
+    // update in response to deps change.
+    const [cursor, data] = Tracker.nonreactive(() => {
+      const cursor = factory()
+      return [ cursor, cursor.fetch() ]
+    })
+    dispatch({
+      type: 'refresh',
+      data: data
+    })
+
+    const observer = cursor.observe({
+      addedAt (document, atIndex, before) {
+        dispatch({ type: 'addedAt', document, atIndex })
+      },
+      changedAt (newDocument, oldDocument, atIndex) {
+        dispatch({ type: 'changedAt', document: newDocument, atIndex })
+      },
+      removedAt (oldDocument, atIndex) {
+        dispatch({ type: 'removedAt', atIndex })
+      },
+      movedTo (document, fromIndex, toIndex, before) {
+        dispatch({ type: 'movedTo', fromIndex, toIndex })
+      },
+      // @ts-ignore
+      _suppress_initial: true
+    })
+
+    return () => {
+      observer.stop()
+    }
+  }, deps)
+
+  return data
+}
+
+const useFindServer = <T = any>(factory: () => Mongo.Cursor<T>, deps: DependencyList) => Tracker.nonreactive(() => factory().fetch())
```

Comment:
The code in the block ran in the `useReducer` lazy load should be shared here/

Reply:
They currently do the same thing. Not sure what you mean.

Reply:
This was in reference to the additional logic I've suggested for validating that we have a cursor, the server will need it too.